hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/HederaState.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.state;
+
+import com.hedera.node.app.spi.state.ReadableStates;
+import com.hedera.node.app.spi.state.WritableStates;
+
+public interface HederaState {
+    ReadableStates createReadableStates(String serviceName);
+
+    WritableStates createWritableStates(String serviceName);
```

Comment:
Maybe we can create an enum for all services in this module. Since this depends on all services anyway it knows the list of services.

Reply:
This is a good question. My original design was just like that, that the application would know exactly the names and types of all services. And for now, maybe that is the way to go. But I have in my mind a hope that we can even abstract that, so that we can discover at runtime which services exist and load them, without having to have special knowledge of them.

The reason this may be important, is there may be variations of Hedera networks. Maybe a CBDC network will have some additional services, or maybe some fewer services, than the public network. Or maybe a private network will have its own collection of services. If each such variation has to have a fork of the app module, it will be more costly to maintain than if they just deploy the services they want and the app module can automatically wire things up.

So I went with a String with the open that it will be dynamic at some point.